sys/net/gnrc/sock/gnrc_sock.c

```diff
@@ -224,6 +224,25 @@ ssize_t gnrc_sock_recv(gnrc_sock_reg_t *reg, gnrc_pktsnip_t **pkt_out,
     return 0;
 }
 
+/* find the last extension header, if present */
+static gnrc_pktsnip_t *_last_ext_hdr(gnrc_pktsnip_t *ipv6)
+{
+#ifdef MODULE_GNRC_IPV6_EXT_OPT
+    if (ipv6->next == NULL || ipv6->next->type != GNRC_NETTYPE_IPV6_EXT) {
+        return NULL;
+    }
+
+    while (ipv6->next && ipv6->next->type == GNRC_NETTYPE_IPV6_EXT) {
+        ipv6 = ipv6->next;
+    }
+
+    return ipv6;
+#else
+    (void)ipv6;
+    return NULL;
+#endif
```

Comment:
this looks like the if is not needed

i think 
```suggestion
static gnrc_pktsnip_t *_last_ext_hdr(gnrc_pktsnip_t *ipv6)
{
#ifdef MODULE_GNRC_IPV6_EXT_OPT
    while (ipv6->next && ipv6->next->type == GNRC_NETTYPE_IPV6_EXT) {
        ipv6 = ipv6->next;
    }

    return ipv6;
#else
    return ipv6;
#endif
```
    while (ipv6->next && ipv6->next->type == GNRC_NETTYPE_IPV6_EXT) {
        ipv6 = ipv6->next;
    }

    return ipv6;
```

would fit and also avoid the null case handlng in the function using this thing

Reply:
But then it would return the IPv6 header as the last extension header when it should return none.

Reply:
but if none is retruned the function using this is using the ipv6 header

Reply:
Yes but `ipv6_ext_t` != `ipv6_hdr_t`, so instead of `nh` we would overwrite `v_tc_fl`

Reply:
`GNRC_NETTYPE_IPV6_EXT` is only defined if `gnrc_ipv6_ext_opt` is used so we need an extra `#ifdef`